services/sso/src/main/java/software.amazon.awssdk.services.sso/internal/SsoProfileCredentialsProviderFactory.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sso.internal;
+
+import static software.amazon.awssdk.services.sso.internal.SsoCachedTokenFileUtils.generateCachedTokenPath;
+import static software.amazon.awssdk.utils.UserHomeDirectoryUtils.userHomeDirectory;
+
+import java.nio.file.Paths;
+import java.util.function.Supplier;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkTestInternalApi;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.SsoCredentialsProviderFactory;
+import software.amazon.awssdk.auth.credentials.internal.SsoCredentialsProfileProperties;
+import software.amazon.awssdk.profiles.Profile;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sso.SsoClient;
+import software.amazon.awssdk.services.sso.auth.SsoCredentialsProvider;
+import software.amazon.awssdk.services.sso.model.GetRoleCredentialsRequest;
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+/**
+ * An implementation of {@link SsoCredentialsProviderFactory} that allows users to get SSO role credentials using the startUrl
+ * sprcified in either a {@link Profile} or environment variables.
+ */
+@SdkInternalApi
+public class SsoProfileCredentialsProviderFactory implements SsoCredentialsProviderFactory {
```

Comment:
Is this what's supposed to be in that string I commented on earlier? If so, should this be a protected API, since its constructor is invoked by the auth module?

Reply:
Agree, should I move it out of the internal package?

Reply:
Yeah, if it's protected it shouldn't be in the internal package.